doc/http_api.md

```diff
@@ -0,0 +1,108 @@
+# HTTP API
+
+Enables mods to make HTTP requests.
+
+## Settings
+
+To use this api, mods need to be listed in one or the other following 
+settings in csv format:
+* `secure.trusted_mods`
+* `secure.http_mods`
+
+An example would be:
+
+```
+secure.http_mods = modname1, modnametwo, modenamethree
+```
+
+## Types
+
+### HttpRequest
+
+A table consisting of the following paramaters:
+
+* url (required): 
+  * `http://subdomain.domain.tld`
+* timeout (optional): 
+  * Time in seconds
+  * Default is what `curl_timeout` setting is set to (NOTE: setting is 
+    in milliseconds)
+* method (optional):
+  * Supports `GET`, `POST`, `PUT`, `DELETE`
+  * Default is `GET`
+* data (optional):
+  * Supports string or table
+  * Tables are encoded as `x-www-form-urlencoded` key value pairs
+* post_data (optional):
+  * Deprecated, see data above
+* user_agent (optional):
+  * Replaces the default user agent if provided
+* extra_headers (optional):
+  * Table of strings
+  * Each string needs to follow http spec, `"key: value"`
+* multipart (optional):
+  * Boolean, default false
+  * Method must be `POST`
+  * [see here for more info](https://en.wikipedia.org/wiki/MIME#Multipart_messages)
```

Comment:
Text of a link should be a bit unique to the link for readability, some screen reader users will use keyboard shortcuts to read the text of each link. If all links have similar text, it's hard to use.

```suggestion
  * Learn more at [MIME (Wikpedia)](https://en.wikipedia.org/wiki/MIME#Multipart_messages)
```